src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -341,7 +341,7 @@ export class KernelProcess implements IKernelProcess {
             traceInfo(`Launching Raw Kernel & not daemon ${this.launchKernelSpec.display_name} # ${executable}`);
             const [executionService, env] = await Promise.all([
                 this.processExecutionFactory.create(this.resource),
-                this.kernelEnvVarsService.getEnvironmentVariables(this.resource, this.launchKernelSpec)
+                this.kernelEnvVarsService.getEnvironmentVariables(this.resource, undefined, this.launchKernelSpec)
```

Comment:
Interpreter not possible in this situation since this is a non python launch. I'll add a comment.

Reply:
Un-resolving comment here. This looks correct here, but the reading is a bit funny (to me) since the daemon service is not always supported. However for the non-deamon cases the PythonKernelLauncherDaemon does return a non-daemon exeObs, so this here is the non-interpreter case.

Reply:
> in this situation since this is a non python launch. I'll add a comment.

This isn't correct, we need the interpreter here
If launching a java kernel from a conda environment, we need the associated conda environment to be actiavted.

That's whats happening here https://github.com/microsoft/vscode-jupyter/pull/6518/files#diff-c2b62e495d0dabb0ddd0d2418166b8d017dc981428b3916a3ea8e989eeb6a803L43